### ISSUE TITLE
Account expiration UI is broken

### ISSUE THREAD

**Issue by estellecomment** [tchapgouv/tchap-web-v4#959]
Maybe linked in case of problem : #423 (if no problem, maybe just close it?)
See also : https://github.com/tchapgouv/tchap-product/issues/288

- [x] warning dialog appears when account is expired, without page reload
- [ ] button "send a new email" works
  - [x] email is sent
  - [x] visual feedback confirms that email is sent
    - [ ] and uses familiar colors and visual components
  - [ ] email link works for re-enabling
  - [x] clicking the button twice sends two emails
    - [x] and user can see confirmation that the second click did something 
- [ ] button "I clicked the link" works
  - [ ] when link has really been clicked, user sees confirmation, and can now send a message
  - [x] when link has not yet been clicked, user sees explanation, and is not re-enabled
- [ ] when user logs into an expired account
  - [x] user sees dialog
    - [x] yes, but requires page reload
    - [x] yes, without needing page reload
  - [ ] user can renew their account
  - [ ] user can give up renewing account, and log out

- [x] unit tests! (move this to a separate issue if needed, to finish the prod code within the sprint)
  - [x] ExpiredAccountHandler
    - [x] opens dialog if the account expires
    - ~~handler is properly registered on app startup~~ too much work to mock out dependencies in that startup code ! Not doing it.
  - [x] ExpiredAccountDialog
    - [x] sends email when link is clicked
    - [x] does not send email if it's too soon
    - [x] when "I renewed" is clicked, and the account is renewed, it displays "cool" and unblocks user
    - [x] when "I renewed" is clicked, and the account is not renewed, it displays "not cool" and keeps dialog up

**Comment by MarcWadai** [comment]
Web tests have been done on this ticket https://github.com/tchapgouv/tchap-infra/issues/2574

**Comment by estellecomment** [comment]
Le backend pour l'expiration de compte a l'air d'avoir du travail en cours, non ? Si les API ont pas changé alors on est bon, c'est pas clair pour moi.

**Comment by estellecomment** [comment]
Our code needs updating : the API for matrixclient has changed and the code is broken.

**Comment by estellecomment** [comment]
Closing this without testing the actual renewal : it's more of a backend problem and will be tested when we roll out the feature in the backend.
